File: tests/fixtures/codegen/dimensions/pointers-expected/queries/arrays/optionalflexible.go
package arrays

import (
	"context"
	pgsidpgx "example.com/pgsid-validation/generated/queries/pgsid/pgx"
	"github.com/jackc/pgx/v5/pgtype"
)

const OptionalFlexibleSQL = "SELECT optional_flexible FROM arrays;"

type OptionalFlexibleParams struct {
}
type OptionalFlexibleRow struct {
	OptionalFlexible *pgtype.Array[int32] `db:"optional_flexible"`
}

func (q *Queries) OptionalFlexible(ctx context.Context) (OptionalFlexibleRow, error) {
	var row OptionalFlexibleRow
	err := q.db.QueryRow(ctx, OptionalFlexibleSQL).Scan(pgsidpgx.ArrayTarget[int32](&row.OptionalFlexible, []int{1, 2}, "OptionalFlexible", "optional_flexible", true))
	return row, err
}

File: tests/fixtures/codegen/dimensions/structs-expected/typescript/runtime/queries/arrays/OptionalFlexible.ts
import type { OptionalFlexibleParams, OptionalFlexibleRow } from "../../../types/queries/arrays/OptionalFlexible.js";
export type { OptionalFlexibleParams, OptionalFlexibleRow } from "../../../types/queries/arrays/OptionalFlexible.js";
import type * as pgsid from "./pgsid/queryable.js";
export const optionalFlexibleSql = "SELECT optional_flexible FROM arrays;";
export async function optionalFlexible(db: pgsid.Queryable): Promise<OptionalFlexibleRow | undefined> {
    const result = await db.query(optionalFlexibleSql, []);
    const row = result.rows[0];
    if (row === undefined)
        return undefined;
    return row as OptionalFlexibleRow;
}
